src/index.ts

```diff
@@ -563,6 +612,26 @@ async function prepareTxs({
 				),
 			}
 		}
+	} else if (linkDetails.tokenType == 2) {
+		VERBOSE && console.log('checking ERC721 allowance...')
+		try {
+			const approveTx = await prepareApproveERC721Tx(
+				address,
+				String(linkDetails.chainId),
+				linkDetails.tokenAddress!,
+				linkDetails.tokenId)
+
+			approveTx && unsignedTxs.push(approveTx)
+		} catch (error) {
+			console.error(error)
+			return {
+				unsignedTxs: [],
+				status: new interfaces.SDKStatus(
+					interfaces.EPrepareCreateTxsStatusCodes.ERROR_PREPARING_APPROVE_ERC721_TX,
+					'Error preparing the approve ERC721 tx, please make sure you have enough balance and have approved the contract to spend your tokens'
```

Comment:
I think error message doesn't need the balance comment - approval should still go through even if the address doesn't have that nft Id in posession (I think?)